dbschema/product.esdl

```diff
@@ -0,0 +1,108 @@
+module default {
+  abstract type Product extending Engagement::Child {
+    overloaded engagement: LanguageEngagement;
+
+    scriptureReferences: Scripture::Collection;
+
+    multi mediums: Product::Medium;
+    multi purposes: Product::Purpose;
+    multi steps: Product::Step;
+    methodology: Product::Methodology;
+
+    describeCompletion: str;
+    placeholderDescription: str;
+    pnpIndex: int16;
+    progressTarget: int16;
+    progressStepMeasurement: Product::ProgressMeasurement;
+
+    #  createdAt: cal::local_date;  #TODO - is this desired here?  Already exists in Mixin::Timestamped?
+
+    #TODO - category := ??? - add this computed field here after migration
+  }
+}
+
+module Product {
+  type DirectScriptureProduct extending default::Product {
+    unspecifiedScripture: Scripture::Collection;  #TODO - is this needed?  Or of a different type?
+
+    #TODO - also below, should these two be in their own type and inherited?
+    totalVerses: int16;  #TODO - computed?
+    totalVerseEquivalents: int16;  #TODO - computed?
```

Comment:
They could be but their calculations are different. 
Direct has the `unspecifiedScripture` to include.
Derivative has the `override` to consider over the `.produces.scripture`.

`totalVerses` could be calculated by DB fairly easily.
```edgeql
range_get_upper() - range_get_lower() # & sum
```

`totalVerseEquivalents` is harder because there's the book factor that the DB doesn't know about.
https://github.com/SeedCompany/cord-api-v3/blob/f8acf9130ada9dfce175787f6bc6a350fcc3217a/src/components/scripture/book-difficulty-factor.ts

If we encoded the difficult enum or factor in our scripture types, this would be possible.

Currently the app calculates this and writes to DB. So possible that we continue down that route for now...

Reply:
Well, let me propose that we just put  the `totalVerseEquivalents` as a `TODO` for now following the "Migrate first" approach.

Which means that we'd only have `totalVerses` in a new subtype, so let's just punt that idea until later as well.

Reply:
Yeah agreed. The app can continue to calculate and save manually to DB. These are one of the first things to be re-worked after migration anyways.